### PR TITLE
[AutoML] Enable style cop rules & resolve errors

### PR DESCRIPTION
**Pull request by daholste**

### CODE REVIEW

File: src/mlnet/ProgressBar/ProgressBarBase.cs [hunk 1]
```diff
@@ -16,20 +16,22 @@ static ProgressBarBase()
             Encoding.RegisterProvider(CodePagesEncodingProvider.Instance);
         }
 
-        protected readonly DateTime _startDate = DateTime.Now;
+        public readonly DateTime StartDate;
```
Comment (Dmitry-A) [inline src/mlnet/ProgressBar/ProgressBarBase.cs]: not a big deal either way but I don't see it -- why this had to switch to pubic?

Comment (daholste) [reply src/mlnet/ProgressBar/ProgressBarBase.cs]: Apparently, there was one usage outside the ProgressBarBase inheritance chain

File: src/Microsoft.ML.AutoML/API/ColumnInference.cs [hunk 2]
```diff
@@ -69,24 +70,37 @@ public sealed class ColumnInformation
...
         /// </summary>
-        public ICollection<string> TextColumnNames { get; } = new Collection<string>();
+        /// <value>The default value is a new, empty <see cref="Collection{String}"/>.</value>
+        public ICollection<string> TextColumnNames { get; }
 
         /// <summary>
         /// The dataset columns that AutoML should ignore.
         /// </summary>
-        public ICollection<string> IgnoredColumnNames { get; } = new Collection<string>();
+        /// <value>The default value is a new, empty <see cref="Collection{String}"/>.</value>
```
Comment (sharwell) [inline src/Microsoft.ML.AutoML/API/ColumnInference.cs]: 📝 This should really be `Collection{T}`. See dotnet/csharplang#401.

File: src/Microsoft.ML.AutoML/API/ExperimentSettings.cs [hunk 3]
```diff
@@ -33,28 +34,42 @@ public abstract class ExperimentSettings
...
+
+        internal int MaxModels;
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="ExperimentSettings"/>.
+        /// </summary>
+        public ExperimentSettings()
+        {
+            MaxExperimentTimeInSeconds = 24 * 60 * 60;
+            CancellationToken = default;
```
Comment (sharwell) [inline src/Microsoft.ML.AutoML/API/ExperimentSettings.cs]: :bulb:

1. This is unnecessary. The property is going to get the default value anyway.
2. Prefer explicit `CancellationToken.None` to `default` when assigning to `CancellationToken`.

File: src/Microsoft.ML.AutoML/Sweepers/SmacSweeper.cs [hunk 4]
```diff
@@ -328,21 +341,21 @@ private double[][] GetForestRegressionLeafValues(FastForestRegressionModelParame
...
         }
 
-        // Todo: Remove the reflection below for TreeTreeEnsembleModelParameters methods GetLeaf and GetLeafValue. 
+        // Todo: Remove the reflection below for TreeTreeEnsembleModelParameters methods GetLeaf and GetLeafValue.
         // Long-term, replace with tree featurizer once it becomes available
         // Tracking issue -- https://github.com/dotnet/machinelearning-automl/issues/342
-        private static MethodInfo GetLeafMethod = typeof(TreeEnsembleModelParameters).GetMethod("GetLeaf", BindingFlags.NonPublic | BindingFlags.Instance);
-        private static MethodInfo GetLeafValueMethod = typeof(TreeEnsembleModelParameters).GetMethod("GetLeafValue", BindingFlags.NonPublic | BindingFlags.Instance);
+        private static MethodInfo _getLeafMethod = typeof(TreeEnsembleModelParameters).GetMethod("GetLeaf", BindingFlags.NonPublic | BindingFlags.Instance);
+        private static MethodInfo _getLeafValueMethod = typeof(TreeEnsembleModelParameters).GetMethod("GetLeafValue", BindingFlags.NonPublic | BindingFlags.Instance);
```
Comment (sharwell) [inline src/Microsoft.ML.AutoML/Sweepers/SmacSweeper.cs]: 💡 Prefer `static readonly` here, at which point the rename is not required.

File: src/Microsoft.ML.AutoML/Sweepers/SweeperBase.cs [hunk 5]
```diff
@@ -19,11 +19,14 @@ internal abstract class SweeperBase : ISweeper
...
-            //[Argument(ArgumentType.Multiple, HelpText = "Swept parameters", ShortName = "p", SignatureType = typeof(SignatureSweeperParameter))]
             public IValueGenerator[] SweptParameters;
 
-            //[Argument(ArgumentType.LastOccurenceWins, HelpText = "Number of tries to generate distinct parameter sets.", ShortName = "r")]
-            public int Retries = 10;
+            public int Retries;
+
+            public ArgumentsBase()
+            {
+                Retries = 10;
```
Comment (sharwell) [inline src/Microsoft.ML.AutoML/Sweepers/SweeperBase.cs]: 💭 Why are all the initializations moving to constructors?

File: src/mlnet/Telemetry/DotNetAppInsights/EnvironmentProvider.cs [hunk 6]
```diff
@@ -11,10 +11,10 @@ namespace Microsoft.DotNet.Cli.Utils
 {
     public class EnvironmentProvider : IEnvironmentProvider
     {
-        private static char[] s_pathSeparator = new char[] { Path.PathSeparator };
-        private static char[] s_quote = new char[] { '"' };
+        private static char[] _sPathSeparator = new char[] { Path.PathSeparator };
```
Comment (sharwell) [inline src/mlnet/Telemetry/DotNetAppInsights/EnvironmentProvider.cs]: 💡 `_pathSeparator`

File: src/mlnet/Telemetry/DotNetAppInsights/EnvironmentProvider.cs [hunk 7]
```diff
@@ -11,10 +11,10 @@ namespace Microsoft.DotNet.Cli.Utils
...
+        private static char[] _sQuote = new char[] { '"' };
```
Comment (sharwell) [inline src/mlnet/Telemetry/DotNetAppInsights/EnvironmentProvider.cs]: 💡 `_quote`

File: src/mlnet/Telemetry/DotNetAppInsights/StreamForwarder.cs [hunk 8]
```diff
@@ -11,8 +11,8 @@ namespace Microsoft.DotNet.Cli.Utils
 {
     public sealed class StreamForwarder
     {
-        private static readonly char[] s_ignoreCharacters = new char[] { '\r' };
-        private static readonly char s_flushBuilderCharacter = '\n';
+        private static readonly char[] _sIgnoreCharacters = new char[] { '\r' };
```
Comment (sharwell) [inline src/mlnet/Telemetry/DotNetAppInsights/StreamForwarder.cs]: 💡 `_ignoreCharacters`

File: src/mlnet/Telemetry/DotNetAppInsights/StreamForwarder.cs [hunk 9]
```diff
@@ -11,8 +11,8 @@ namespace Microsoft.DotNet.Cli.Utils
...
+        private static readonly char _sFlushBuilderCharacter = '\n';
```
Comment (sharwell) [inline src/mlnet/Telemetry/DotNetAppInsights/StreamForwarder.cs]: :bulb: `_flushBuilderCharacter`

**Pull request merged**